### ISSUE TITLE
ST_Union_Aggr exception

### ISSUE THREAD

**Issue by beautifulpython** [zilliztech/arctern#138]
I got an exception when run my test code below :
```python
from pyspark.sql import SparkSession
from zilliz_pyspark import register_funcs

def run_st_union(spark):
    test_df = spark.read.json("/xxx/st_union.json").cache()
    test_df.createOrReplaceTempView("st_union")
    register_funcs(spark)
    spark.sql("select ST_Union_Aggr_UDF(geos) from (select ST_PolygonFromEnvelope_UDF(a,c,b,d) as geos from st_union) as foo").show(100,0)

#main here.
```
 st_union.json is just like : 
```json
{"a": 13.9, "c": 82.2, "b": 19.1, "d": 83.4}
{"a": 10.1, "c": 91.9, "b": 19.7, "d": 98.3}
{"a": 16.1, "c": 93.3, "b": 16.6, "d": 94.0}
{"a": 11.0, "c": 88.3, "b": 18.7, "d": 98.2}
{"a": 13.9, "c": 82.2, "b": 19.1, "d": 83.4}
{"a": 12.0, "c": 81.5, "b": 16.2, "d": 90.6}
{"a": 10.4, "c": 87.5, "b": 11.7, "d": 92.2}
{"a": 15.5, "c": 88.7, "b": 18.6, "d": 98.4}
{"a": 14.8, "c": 83.0, "b": 16.9, "d": 85.6}
{"a": 10.8, "c": 83.9, "b": 16.5, "d": 84.4}
{"a": 12.5, "c": 80.8, "b": 14.8, "d": 97.1}
```
The messege is :
``` 
ERROR 1: TopologyException: Input geom 0 is invalid: Self-intersection at or near point 14.899999999999999 95.099999999999994 at 14.899999999999999 95.099999999999994
20/02/29 15:43:16 ERROR Executor: Exception in task 0.0 in stage 3.0 (TID 3)
```

postgis test :
```sql
sql :
drop table t1;
create table t1 (a real,c real,b real,d real);
insert into t1 values 
(10.1,91.9,19.7,98.3),
(16.1,93.3,16.6,94.0),
(11.0,88.3,18.7,98.2),
(13.9,82.2,19.1,83.4),
(12.0,81.5,16.2,90.6),
(10.4,87.5,11.7,92.2),
(15.5,88.7,18.6,98.4),
(14.8,83.0,16.9,85.6),
(10.8,83.9,16.5,84.4),
(12.5,80.8,14.8,97.1)
;
select st_astext(st_union(geo)) from (select st_makeEnvelope(a,c,b,d) as geo from t1) as foo;

result :
 POLYGON((16.8999996185303 83.4000015258789,19.1000003814697 83.4000015258789,19.1000003814697 82.1999969482422,16.2000007629395 82.1999969482422,16.2000007629395 81.5,14.8000001907349 81.5,14.8000001907349 80.8000030517578,12.5 80.8000030517578,12.5 81.5,12 81.5,12 83.9000015258789,10.8000001907349 83.9000015258789,10.8000001907349 84.4000015258789,12 84.4000015258789,12 88.3000030517578,11.6999998092651 88.3000030517578,11.6999998092651 87.5,10.3999996185303 87.5,10.3999996185303 91.9000015258789,10.1000003814697 91.9000015258789,10.1000003814697 98.3000030517578,15.5 98.3000030517578,15.5 98.4000015258789,18.6000003814697 98.4000015258789,18.6000003814697 98.30
00030517578,19.7000007629395 98.3000030517578,19.7000007629395 91.9000015258789,18.7000007629395 91.9000015258789,18.7000007629395 88.3000030517578,16.2000007629395 88.
3000030517578,16.2000007629395 85.5999984741211,16.8999996185303 85.5999984741211,16.8999996185303 83.4000015258789))
```

**Comment by nezha2017** [comment]
this bug should be fix in PR #137 ,please update the arctern and retry it, @beautifulpython

**Comment by beautifulpython** [comment]
> this bug had been fixed in PR #137 ,please update the arctern and retry it, @beautifulpython

I merged latest version,but still not work.The same ex is threw.
```scala
spark.sql("select ST_Union_Aggr_UDF(ST_PolygonFromEnvelope_UDF(1,2,3,4))").show(100,0)  // ok
spark.sql("select ST_Union_Aggr_UDF(ST_PolygonFromEnvelope_UDF(a,c,b,d)) from st_union").show(100,0) //error
```
